auth.go

```diff
@@ -12,88 +13,82 @@ import (
 // TODO: add interceptors to provide "role required" and "group
 // member" using gimlet.Authenticator
 
-func MakeAuthenticationRequiredUnaryInterceptor(um gimlet.UserManager, conf gimlet.UserMiddlewareConfiguration) grpc.UnaryServerInterceptor {
+func MakeAuthenticationRequiredUnaryInterceptor(um gimlet.UserManager, conf gimlet.UserMiddlewareConfiguration, ignore ...string) grpc.UnaryServerInterceptor {
+	ignoreMap := getIgnoreMap(ignore)
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-		meta, ok := metadata.FromIncomingContext(ctx)
-		if !ok {
-			return nil, grpc.Errorf(codes.Unauthenticated, "missing metadata from context")
+		if info == nil || !ignoreMap[info.FullMethod] {
+			var err error
+			if ctx, err = checkUser(ctx, um, conf); err != nil {
```

Comment:
Is it actually necessary to set the user in the context for the unary RPCs? It seems like the stream RPCs don't need it.

Reply:
I don't think so. I feel like it is safe to remove that, what do you think?

Reply:
Eh, we can remove it if we want to some other time.